### DISCUSSION TITLE
Does SkiaSharp work with Wasi Console Apps (WASM)?

### DISCUSSION THREAD

**Discussion by enghch** [mono/SkiaSharp#3009] [Q&A]
I have a Wasi Console App (.NET 9, preview 6) which compiles to a single .wasm file. I can add nuget references to SkiaSharp and the SkiaSharp.NativeAssets,WebAssembly and it appears to compile file. But when I try to call Skia methods from my code, I get a DllNotFoundException for libSkiaSharp.

`"System.TypeInitializationException: TypeInitialization_Type, SkiaSharp.SKImageInfo --- System.DllNotFoundException: libSkiaSharp   at SkiaSharp.SKImageInfo..cctor()`

SkiaSharp is supported in Blazor Wasm so I assumed it would work in Wasi Wasm. Is there some extra build step I need to add to get the native assets statically linked into my wasm file?

**Comment by ToolmakerSteve** [comment]
> SkiaSharp is supported in Blazor Wasm

Have you tested this in Blazor, yourself?

See this [open SkiaSharp / Blazor Wasm issue](https://github.com/mono/SkiaSharp/issues/2745). 

It links to a closed issue, which contains a comment with a code snippet that might be relevant.:  
https://github.com/mono/SkiaSharp/issues/3037#issuecomment-2419406988

Perhaps this line:
```
<NativeFileReference Include="$(SkiaSharpStaticLibraryPath)\3.1.56\st\*.a" />
```

You could follow the steps (and these issues); get SkiaSharp working with Blazor Wasm. Then compare the resulting csproj to yours.

**Reply by mattleibow** [reply]
Not WASI, but a blazor-less WASM sample: https://github.com/mono/SkiaSharp/tree/main/samples/Basic/BrowserWebAssembly

Let us know if you manage to get it all working!

**Comment by tossnet** [comment]
SkiaSharp and Blazor Wasm  has never worked for me :'( 

![image](https://github.com/user-attachments/assets/f505e930-39d3-41a4-8e18-3c074b7afc0d)

**Reply by mattleibow** [reply]
Are you able to open a new issue with this? I am pretty sure I fixed all of the .NET 9 issues in the last few releases. But, maybe I missed something.

**Reply by tossnet** [reply]
ok if you want I create new issue .
I was just testing your sample here [https://github.com/mono/SkiaSharp/blob/main/samples/Basic/BlazorWebAssembly/SkiaSharpSample/SkiaSharpSample.csproj ](https://github.com/mono/SkiaSharp/blob/main/samples/Basic/BlazorWebAssembly/SkiaSharpSample/SkiaSharpSample.csproj )but the problem with this example is that it's not autonomous and depends on many other sources (`<Import Project=”..\..\..\..\..\`)
